File: app/src/main/java/com/lua/custrecycleradapter/LuaCustRecyclerHolder.java
package com.lua.custrecycleradapter;

import android.view.View;
import androidx.recyclerview.widget.RecyclerView;

public class LuaCustRecyclerHolder extends RecyclerView.ViewHolder {
    public View view;

    public LuaCustRecyclerHolder(View view) {
        super(view);
        this.view = view;
    }

    // 添加释放资源的方法
    public void release() {
        if (view != null) {
            view = null;
        }
    }
}

File: app/src/main/java/github/znzsofficial/adapter/LuaCustRecyclerHolder.kt
package github.znzsofficial.adapter

import android.view.View
import androidx.recyclerview.widget.RecyclerView

open class LuaCustRecyclerHolder(itemView: View?) : RecyclerView.ViewHolder(itemView!!) {
    var views: Any? = null

    fun setTag(tag:Any){
        views = tag
    }
    fun getTag() = views

    // 添加释放资源的方法
    open fun release() {
        views = null
        // 释放itemView中的资源
        if (itemView is Releasable) {
            (itemView as Releasable).release()
        }
    }
}

// 可释放资源的接口
interface Releasable {
    fun release()
}